clang/lib/Sema/SemaLambda.cpp

```
@@ -2332,6 +2332,20 @@ ExprResult Sema::BuildLambdaExpr(SourceLocation StartLoc,
     maybeAddDeclWithEffects(LSI->CallOperator);
   }
 
+  // This is for GCC compatibility. If any lambda captures are actually used in the
+  // function body. GCC silently removes the naked attribute when captures are
```

Regardless of what gcc does here, silently ignoring the naked attribute is not acceptable: the result will almost certainly be broken.

Probably simplest to just reject any usage of naked with captures; if gcc's behavior is broken, probably nobody is using it.